src/usePermission.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+import { off, on } from './util';
+
+type PermissionDesc =
+  | PermissionDescriptor
+  | DevicePermissionDescriptor
+  | MidiPermissionDescriptor
+  | PushPermissionDescriptor;
+
+const noop = () => {};
+
+const usePermission = (permissionDesc: PermissionDesc): string => {
```

Comment:
Instead of `string` return type, maybe we can make it more specific?

```ts
'' | 'granted' | 'denied' | 'prompt'
```

Reply:
Thanks for your suggestion.